src/components/organisms/AuthenticationModal/LoginForm/LoginForm.tsx

```diff
@@ -56,31 +61,43 @@ const LoginForm: React.FunctionComponent<PropsType> = ({
   const onSubmit = async (data: LoginFormData) => {
     if (!venue) return;
     try {
-      if (venue.requiresTicketCode) await axios.get(codeCheckUrl(data.code));
-      if (venue.requiresEmailVerification)
-        await axios.get(codeCheckUrl(data.email));
-
-      const auth = await signIn(data);
-
-      if (
-        auth.user &&
-        (venue.requiresTicketCode || venue.requiresEmailVerification)
-      ) {
-        firebase
-          .firestore()
-          .doc(`userprivate/${auth.user.uid}`)
-          .get()
-          .then((doc) => {
-            if (auth.user && doc.exists) {
-              firebase
-                .firestore()
-                .doc(`userprivate/${auth.user.uid}`)
-                .update({
-                  codes_used: [...(doc.data()?.codes_used || []), data.code],
-                  date_of_birth: data.date_of_birth,
-                });
-            }
-          });
+      await signIn(data);
+
+      if (venue.access?.includes(VenueAccessMode.Codes)) {
```

Comment:
I think I saw that we changed the type for `venue.access` somewhere to no longer be an array, but this code seems to imply that it relies on it being an array?

Reply:
Nice catch will change it.